tt_metal/include/compute_kernel_api/transpose_wh.h

```diff
@@ -65,14 +87,27 @@ ALWI void transpose_wh_init_short(uint32_t icb) {
  */
  // clang-format on
 ALWI void transpose_wh_tile(uint32_t icb, uint32_t itile, uint32_t idst) {
-    UNPACK((
-#ifdef ARCH_GRAYSKULL
-        llk_unpack_A<BroadcastType::NONE, false>(icb, itile, true)
-#else
-        llk_unpack_A<BroadcastType::NONE, false>(icb, itile, false)
+#if defined(TRISC_MATH) || defined(TRISC_UNPACK)
+    const std::uint32_t src_format = get_operand_src_format(icb);
+    const bool is_32bit = (src_format & 0xf) == (std::uint32_t)DataFormat::Int32;
+
+    if (is_32bit) {
+        copy_tile_init(icb);
+        copy_tile(icb, itile, idst);
+
+        transpose_wh_dest_init_short();
+        transpose_wh_dest<true>(idst);
```

Comment:
@nvelickovicTT This is the key change that might be causing issues when combined with tilize/untilize?